app/antedecorators/gas.go

```diff
@@ -1,82 +1,19 @@
 package antedecorators
 
 import (
-	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
 	"github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkacltypes "github.com/cosmos/cosmos-sdk/types/accesscontrol"
-	aclkeeper "github.com/cosmos/cosmos-sdk/x/accesscontrol/keeper"
-	acltypes "github.com/cosmos/cosmos-sdk/x/accesscontrol/types"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
 )
 
-const (
-	GasMultiplierNumerator                   uint64 = 1
-	DefaultGasMultiplierDenominator          uint64 = 1
-	WasmCorrectDependencyDiscountDenominator uint64 = 2
-)
-
-func GetGasMeterSetter(aclkeeper aclkeeper.Keeper) func(bool, sdk.Context, uint64, sdk.Tx) sdk.Context {
+func GetGasMeterSetter(pk paramskeeper.Keeper) func(bool, sdk.Context, uint64, sdk.Tx) sdk.Context {
 	return func(simulate bool, ctx sdk.Context, gasLimit uint64, tx sdk.Tx) sdk.Context {
-		if simulate || ctx.BlockHeight() == 0 {
+		if ctx.BlockHeight() == 0 {
```

Comment:
why don't we have simulate specific gas meter behavior? -  we should still have some infinite gas meter that then applies the scaling factor to be able to accurate simulate, right?

Reply:
we should be careful here given that the cachekv is no longer emitting events, and this is a difference in how the store accesses happen

Reply:
I think for the purposes of this change, just to be safe, we should reintroduce the event emissions in cachekv (despite performance consequences) to ensure the parallel dag resource accesses remain accurate. This will need to be updated in sei-cosmos and then reflected here

Reply:
@udpatil for first comment, new infinite multiplier gas meter has been added for simulation